project-koku/koku
# Handling Source.update and Authentication.update messages

Handles Kafka `Source.update` and `Authentication.update` to ensure for Koku-Providers remain in sync with Platform Sources when the user edits a source from the Platform Sources UI 

Functional Testing was done with the pending change from https://github.com/project-koku/koku/pull/1213

**Testing**
1. Add AWS Source with valid, update with invalid ARN, add bucket.  Verify provider creation fails.  Update to correct ARN and verify provider creation is successful
2.  Update name of source from #1
3.  Add AWS Source, shut down koku API server, add bucket, change name of source, start koku-server, verify that provider is created with latest name
4. Create AWS Source, add bucket, shut down koku API server, Edit Source Name, Delete Source, Start API server.  Verify source is successfully removed
5. Create AWS source, shut down API server, add incorrect bucket, start API server.  Verify that koku provider fails creation.  Do Update on bucket with correct bucket.  Verify koku provider is successfully created
6. Create AWS source, add bucket, verify koku provider is created.  Update bucket with invalid bucket.  Verify that update fails.

**Test Results**
[sources_edit_unittest.txt](https://github.com/project-koku/koku/files/3698786/sources_edit_unittest.txt)
